components/tidb_query_vec_executors/src/index_scan_executor.rs

```diff
@@ -232,59 +321,133 @@ impl IndexScanExecutorImpl {
                 return Err(other_err!("Unexpected missing column {}", col_id));
             }
         }
+        Ok(())
+    }
 
-        if self.decode_handle {
-            // For normal index, it is placed at the end and any columns prior to it are
-            // ensured to be interested. For unique index, it is placed in the value.
+    fn extract_columns_from_datum_format(
+        datum: &mut &[u8],
+        columns: &mut [LazyBatchColumn],
+    ) -> Result<()> {
+        for column in columns {
+            if datum.is_empty() {
+                return Err(other_err!("Value is missing some columns"));
+            }
+            let (value, remaining) = datum::split_datum(datum, false)?;
+            column.mut_raw().push(value);
+            *datum = remaining;
+        }
+        Ok(())
+    }
 
-            let handle_val = if tail_len >= 8 {
-                // This is a unique index, and we should look up PK handle in value.
-                self.decode_handle_from_value(&value[value.len() - 8..])?
-            } else {
-                // This is a normal index. The remaining payload part is the PK handle.
-                // Let's decode it and put in the column.
-                self.decode_handle_from_key(&key[key.len() - 9..])?
-            };
-            columns[self.columns_id_without_handle.len()]
-                .mut_decoded()
-                .push_int(Some(handle_val));
+    fn process_unique_common_handle_value(
+        &mut self,
+        key: &[u8],
+        value: &[u8],
+        columns: &mut LazyBatchColumnVec,
+    ) -> Result<()> {
+        let handle_len = ((value[2] as usize) << 8) + value[3] as usize;
```

Comment:
`read_u16_le`?

Reply:
Will it cause panics when value is not long enough?

Reply:
The length of the value is checked in `process_kv_pair`

Reply:
We should use `read_u16` instead.